CompatBot/EventHandlers/UsernameValidationMonitor.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Threading.Tasks;
+using CompatBot.Database;
+using CompatBot.Utils;
+using DSharpPlus;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+using Microsoft.EntityFrameworkCore;
+
+namespace CompatBot.EventHandlers
+{
+    public static class UsernameValidationMonitor
+    {
+        public static async Task OnUserUpdated(UserUpdateEventArgs args)
+        {
+            await UpdateDisplayName(args.Client, () => args.Client.GetMember(args.UserAfter)).ConfigureAwait(false);
+        }
+
+        public static async Task OnMemberUpdated(GuildMemberUpdateEventArgs args)
+        {
+            await UpdateDisplayName(args.Client, () => args.Member).ConfigureAwait(false);
+        }
+
+        public static async Task OnMemberAdded(GuildMemberAddEventArgs args)
+        {
+            await UpdateDisplayName(args.Client, () => args.Member).ConfigureAwait(false);
+        }
+
+        private static async Task UpdateDisplayName(DiscordClient client, Func<DiscordMember> getGuildMember)
```

Comment:
there's no need to accept lambda as an argument here, not only will it incur needless allocations on each call, it's harder to use for no benefit